code/atom.dm

```diff
@@ -700,6 +700,8 @@
 	SHOULD_NOT_OVERRIDE(1)
 	if(SEND_SIGNAL(src,COMSIG_ATTACKBY,W,user, params, is_special))
 		return
+	if(SEND_SIGNAL(W, COMSIG_ITEM_ATTACKBY_PRE, src, user))
+		return
```

Comment:
why is there two and why is pre not before the other one

Reply:
The first signal is called on the atom being attacked, while the second one is called on the item being used for the attackby; they fill different niches. I can definitely move the order of operations, but I'm not sure they can be merged?

Reply:
Yeah I think the pre should just be first then